### PR TITLE
Introduce SPID connection property

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Addresses #657 

Introduces new `SqlConnection`'s readonly property:
**ServerProcessId**: The server process Id (SPID) of the active connection.

cc @shueybubbles

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs [hunk 1]
```diff
@@ -27,14 +27,15 @@ public static class ConnectivityTest
...
-        public static void EnvironmentHostNameTest()
+        public static void EnvironmentHostNameSPIDTest()
         {
             SqlConnectionStringBuilder builder = (new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) { Pooling = true });
             builder.ApplicationName = "HostNameTest";
 
             using (SqlConnection sqlConnection = new SqlConnection(builder.ConnectionString))
             {
                 sqlConnection.Open();
+                int sqlclientSPID = sqlConnection.ServerProcessId;
```
Comment (shueybubbles) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs:38]: isn't this read only guaranteed to be consistent with the below query result _after_ the query has completed?

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs [hunk 2]
```diff
@@ -44,6 +45,8 @@ public static void EnvironmentHostNameTest()
                     sessionSpid = reader.GetInt16(0);
                 }
 
+                Assert.Equal(sqlclientSPID, sessionSpid);
```
Comment (shueybubbles) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs:48]: xunit's Assert.Equal puts "expected" before "actual"

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs (hunk 1 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs:38]: Yes I'll add check there too.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs (hunk 2 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs]: Yes I changed that :)

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx [hunk 3]
```diff
@@ -4560,4 +4560,7 @@
   <data name="SQL_SettingCredentialWithInteractive" xml:space="preserve">
     <value>Cannot set the Credential property if 'Authentication=Active Directory Interactive' has been specified in the connection string.</value>
   </data>
-</root>
+  <data name="SqlConnection_ServerProcessId" xml:space="preserve">
+    <value>Server Process Id (SPID) of the active connection.</value>
+  </data>
+</root>
```
Comment (JRahnama) [inline src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx:4566]: NIT: Add an extra line please.

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 4]
```diff
@@ -505,4 +505,4 @@
   <Import Project="$(NetFxSource)tools\targets\GenerateThisAssemblyCs.targets" />
   <Import Project="$(NetFxSource)tools\targets\GenerateAssemblyRef.targets" />
   <Import Project="$(NetFxSource)tools\targets\GenerateAssemblyInfo.targets" />
-</Project>
+</Project>
```
Comment (JRahnama) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:508]: NIT: Add an extra line please.

File: doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml [hunk 5]
```diff
@@ -1096,6 +1096,11 @@ GO
   
  <see cref="P:Microsoft.Data.SqlClient.SqlConnection.ServerVersion" /> was called while the returned Task was not completed and the connection was not opened after a call to <see cref="M:Microsoft.Data.SqlClient.SqlConnection.OpenAsync(System.Threading.CancellationToken)" />.</exception>
         </ServerVersion>
+      <ServerProcessId>
+        <summary>Gets the server process Id (SPID) of the active connection.</summary>
+        <value>The server process Id (SPID) of the active connection.</value>
+        <remarks>Returns 0 if the connection is inactive on client side.</remarks>
```
Comment (David-Engel) [inline doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml:1102]:
```suggestion
        <remarks>Returns 0 if the connection is inactive on the client side.</remarks>
```

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 6]
```diff
@@ -444,10 +444,10 @@
   </ItemGroup>
   <!-- Resources  -->
   <ItemGroup>
-    <Compile Include="Resources\$(ResxFileName).Designer.cs">
+    <Compile Include="Resources\Strings.Designer.cs">
       <AutoGen>True</AutoGen>
       <DesignTime>True</DesignTime>
-      <DependentUpon>$(ResxFileName).resx</DependentUpon>
+      <DependentUpon>Strings.resx</DependentUpon>
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:450]: Revert? Visual Studio likes to automatically do this for me, too.

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx [hunk 7]
```diff
@@ -4560,4 +4560,7 @@
   <data name="SQL_SettingCredentialWithInteractive" xml:space="preserve">
     <value>Cannot set the Credential property if 'Authentication=Active Directory Interactive' has been specified in the connection string.</value>
   </data>
+  <data name="SqlConnection_ServerProcessId" xml:space="preserve">
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx:4563]: Missing item in SR.resx

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs [hunk 8]
```diff
@@ -10539,6 +10539,15 @@ internal class Strings {
             }
         }
         
+        /// <summary>
+        ///   Looks up a localized string similar to Server Process Id (SPID) of the active connection..
+        /// </summary>
+        internal static string SqlConnection_ServerProcessId {
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs:10545]: Do we need this in sr.designer.cs too?

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx (hunk 7 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx:4563]: It's not desired actually, this is NetFx specific property description.

**Pull request merged**